storagemarket/impl/provider.go

```diff
@@ -289,16 +289,19 @@ func (p *Provider) ImportDataForDeal(ctx context.Context, propCid cid.Cid, data
 	if err != nil {
 		return xerrors.Errorf("failed to create temp file for data import: %w", err)
 	}
+	defer tempfi.Close()
```

Comment:
We have that in `cleanup`.

Reply:
`cleanup` is only called if there's some error.

Reply:
Right, I was confused about that. It seems like it is missing on line 339 though.

Reply:
True.

Reply:
Added missing cleanup call.